sys-kernel/gentoo-kernel-bin/gentoo-kernel-bin-6.6.7.ebuild

```
@@ -110,6 +115,15 @@ src_test() {
 }
 
 src_install() {
+	# Keep the kernel image type we don't want out of install tree
+	local image=$(dist-kernel_get_image_path)
+	if use generic-uki; then
+		# This will get installed in postinst later
+		rm "${BINPKG}/image/usr/src/linux-${KPV}/${image}" || die
+	else
+		rm "${BINPKG}/image/usr/src/linux-${KPV}/${image%/*}/uki.efi" || die
+	fi
```

```suggestion
```suggestion
	# Keep the kernel image type we don't want out of install tree
	local to_remove=$(dist-kernel_get_image_path)
	if ! use generic-uki; then
		to_remove=${to_remove%*}/uki.efi
	fi
	rm "${BINPKG}/image/usr/src/linux-${KPV}/${to_remove}" || die
```
```